CONTRIBUTING.md

```diff
@@ -0,0 +1,45 @@
+# Contributing to leaderboard-backend
+We appreciate your help!
+
+## Before filing an issue
+If you are unsure whether you have found a bug, please consider asking in [our discord](https://discord.gg/TZvfau25Vb) first.
+
+Similarly, if you have a question about a potential feature, [the discord](https://discord.gg/TZvfau25Vb) can be a fantastic resource for first comments.
+
+## Filing issues
+Filing issues is as simple as going to [the issue tracker](https://github.com/speedrun-website/leaderboard-backend/issues), and adding an issue!
+
+## Contributing code
+### Example code contribution flow
+1. Make a fork of this repo.
+1. Name a branch on your fork something descriptive for this change (eg. `UpdateMakefile`).
+1. Commit your changes (Tip! Please read our [Style guide](#style-guide) to help the pull request process go smoothly).
+1. Verify your changes work with `make test`.
+1. Push your branch.
+1. Open a pull request to `speedrun-website/leaderboard-backend`.
+1. Get your pull request approved.
+1. Get someone to click `Rebase and merge`.
+1. Celebrate your amazing changes! 🎉
+
+## Style guide
+### General
+- Be inclusive, this is a project for everyone.
+- Be descriptive, it can be hard to understand abbreviations or short-hand.
+
+### GoLang
+- Comments should be full sentences, starting with a capital letter and ending with punctuation.
```

Comment:
I think this is a place where I would actually relax to a suggestive guideline. Personally, I don't think comments need to be full sentences, so long as they are coherent. Comments end up being this way, but if you have short equations or things of that nature that describe code, full sentences can be clumsy. I would personally change it to say "Comments should be written to be understood by someone who has never seen your code before." Something like that - I'm not solid on anything here.

Reply:
Ah I believe full sentence comments is actually a go standard, to support godoc anyways.
> Notice this comment is a complete sentence that begins with the name of the element it describes. This important convention allows us to generate documentation in a variety of formats, from plain text to HTML to UNIX man pages, and makes it read better when tools truncate it for brevity, such as when they extract the first line or sentence.
-https://blog.golang.org/godoc

Reply:
Ahh, I see here. It seems like this is more for function and package documentation, which is different from general comments. I can agree with that convention for the former two.

Reply:
In general I've found it to be a useful rule, but if people find this super restrictive I am okay changing it.

Reply:
I prefer to keep this, is that okay?

Reply:
I'd love to get someone else to chime in, but if not, I'm not against it enough to push for a change.

Reply:
I've always found the Go standards for committing and commenting to be a bit pedantic and strict, but I think it's worth it in the long run. I'm not excited for a future where we have to block PRs because someone's comment wasn't a sentence, but I imagine we'll reap benefits.

Reply:
Being new to the language I haven't used godoc, but if it's anything like rustdoc I would agree that full sentences and correct grammar is important for any documenting comments. For regular code comments I would go with Josh's suggestion:
> "Comments should be written to be understood by someone who has never seen your code before."